### PR TITLE
Add support for multiple WireGuard interfaces

### PR DESCRIPTION
**Pull request by aptalca**
- Move WireGuard configs to subfolder `/config/wg_confs`
- Attempt to start all `/config/wg_confs/*.conf` in alphabetical order, if any one fails, stop all and delete default route to block network until restart
- Attempt one time migration of active `/config/wg0.conf` to `/config/wg_conf/wg0.conf` to prevent breaking existing installs

### CODE REVIEW

File: root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run [hunk 1]
```diff
@@ -2,10 +2,15 @@
...
-rm -rf /etc/wireguard
-mkdir -p /etc/wireguard
-ln -s /config/wg0.conf /etc/wireguard/wg0.conf
+mkdir -p /config/wg_confs
+
+# migration to subfolder for wg confs
+if [[ -z "$(ls -A /config/wg_confs)" ]] && [[ -f /config/wg0.conf ]]; then
+    echo "**** Performing migration to new folder structure for confs. Please see the image changelog 2023-10-03 entry for more details. ****"
+    cp /config/wg0.conf /config/wg_confs/wg0.conf
+    rm -rf /config/wg0.conf || :
```
Comment (quietsy) [inline root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run:11]: What does `|| :` do?

File: root/etc/s6-overlay/s6-rc.d/svc-wireguard/run [hunk 2]
```diff
@@ -1,4 +1,43 @@
...
+if [[ -z "${WG_CONFS}" ]]; then
+    echo "**** No valid tunnel config found. Please create a valid config and restart the container ****"
+    ip route del default
+    exit 0
+fi
+
+unset FAILED
+for tunnel in ${WG_CONFS[@]}; do
+    echo "**** Activating tunnel ${tunnel} ****"
+    wgquick up "${tunnel}" || ( echo FAILED="${tunnel}" && break)
```
Comment (quietsy) [inline root/etc/s6-overlay/s6-rc.d/svc-wireguard/run:25]: `wg-quick`

File: root/etc/s6-overlay/s6-rc.d/svc-wireguard/run [hunk 3]
```diff
@@ -1,4 +1,43 @@
...
+    declare -p WG_CONFS > /app/activeconfs
+    echo "**** All tunnels are now active ****"
+else
+    echo "**** Tunnel ${FAILED} failed, will stop all others! ****"
+    for tunnel in ${WG_CONFS[@]}; do
+        if [[ "${tunnel}" = "${FAILED}" ]]; then
+            break
+        else
+            echo "**** Disabling tunnel ${tunnel} ****"
+            wgquick down "${tunnel}" || :
```
Comment (quietsy) [inline root/etc/s6-overlay/s6-rc.d/svc-wireguard/run:38]: `wg-quick`

File: root/etc/s6-overlay/s6-rc.d/svc-wireguard/run [hunk 4]
```diff
@@ -1,4 +1,43 @@
...
+        fi
+    done
+    ip route del default
```
Comment (quietsy) [inline root/etc/s6-overlay/s6-rc.d/svc-wireguard/run:41]: What is the effect of this on existing users?

File: root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run [hunk 5]
```diff
@@ -25,7 +30,7 @@ generate_confs () {
         wg genkey | tee /config/server/privatekey-server | wg pubkey > /config/server/publickey-server
     fi
     eval "$(printf %s)
-    cat <<DUDE > /config/wg0.conf
+    cat <<DUDE > /config/wg_confs/wg0.conf
```
Comment (quietsy) [inline root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run:33]: `/app/show-peer` also needs to be changed for the new path

File: root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run (hunk 1 above)
Comment (aptalca) [reply root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run:11]: Makes sure it exits 0 even if the rm operation fails (if the user bind mounted the file, the delete operation will fail)

File: root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run (hunk 5 above)
Comment (aptalca) [reply root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run:33]: I remembered this last night on mobile :)

**Review by quietsy — APPROVED** [review]
LGTM

File: root/etc/s6-overlay/s6-rc.d/svc-wireguard/run [hunk 6]
```diff
@@ -1,4 +1,43 @@
...
+done
+
+if [[ -z "${FAILED}" ]]; then
+    declare -p WG_CONFS > /app/activeconfs
+    echo "**** All tunnels are now active ****"
+else
+    echo "**** Tunnel ${FAILED} failed, will stop all others! ****"
+    for tunnel in ${WG_CONFS[@]}; do
+        if [[ "${tunnel}" = "${FAILED}" ]]; then
+            break
```
Comment (quietsy) [inline root/etc/s6-overlay/s6-rc.d/svc-wireguard/run:35]: Seems that it doesn't like this break when the tunnel fails

Comment (aptalca) [reply root/etc/s6-overlay/s6-rc.d/svc-wireguard/run:38]: It was the other `break` on line 25, which is now fixed

**Review by drizuid — APPROVED** [review]

**Pull request merged**